app/src/main/java/timur/gilfanov/messenger/domain/usecase/user/ObserveAndApplyLocaleUseCase.kt

```diff
@@ -0,0 +1,32 @@
+package timur.gilfanov.messenger.domain.usecase.user
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import timur.gilfanov.messenger.domain.entity.fold
+import timur.gilfanov.messenger.domain.usecase.user.repository.LocaleRepository
+import timur.gilfanov.messenger.util.Logger
+
+class ObserveAndApplyLocaleUseCase(
+    private val observeUiLanguage: ObserveUiLanguageUseCase,
+    private val localeRepository: LocaleRepository,
+    private val logger: Logger,
+) {
+    companion object {
+        private const val TAG = "ObserveAndApplyLocaleUseCase"
+    }
+
+    operator fun invoke(): Flow<Unit> = observeUiLanguage()
+        .distinctUntilChanged()
+        .map { result ->
+            result.fold(
+                onSuccess = { language ->
+                    logger.i(TAG, "Applying locale: $language")
+                    localeRepository.applyLocale(language)
+                },
+                onFailure = { error ->
+                    logger.w(TAG, "Failed to observe language: $error")
+                },
+            )
+        }
```

Comment:
Filtering duplicate values of UI language may lead to skipping of locale update in case if locale for application was changed in system settings for application, not at LanguageScreen. Must remove conflation here.
```suggestion
    operator fun invoke(): Flow<Unit> = observeUiLanguage()
        .map { result ->
            result.fold(
                onSuccess = { language ->
                    logger.i(TAG, "Applying locale: $language")
                    localeRepository.applyLocale(language)
                },
                onFailure = { error ->
                    logger.w(TAG, "Failed to observe language: $error")
                },
            )
        }
```